dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -360,6 +360,7 @@ private boolean shouldDelay() {
         if (delay == null && provider != null) {
             delay = provider.getDelay();
         }
+        this.delay = delay;

```

Comment:
Hi, -1 to this change.

`shouldDelay` is a method that appears read-only. Modifying value within this method does not match with how this method should work.

I'd suggest to get the actual value of delay in `org.apache.dubbo.config.ServiceConfig#export`, or abstract it into a new method.

Reply:
ok , i change it

Reply:
i abstract it into a new method , can you help me see see

Reply:
don't do this, the original change is good enough. The second change is unnecessary and make the code more complex.

Reply:
pls. simply revert 95b394a6033e5a787e70fbdb1a3dd3b0dbc0aeb4. I will follow up.